### ISSUE TITLE
Loss and output functions are different.

### ISSUE THREAD

**Issue by voladorlu** [maciejkula/triplet_recommendations_keras#11]
Hi, It's very nice to see a post of pairwise ranking loss for keras. Could you help me to gain more thinking behind the triplet loss definition? As a new user of keras, it's a little hard to understand why it works since the output of model is "triplet loss", but the compiled model's loss is "identity loss". Please, could you give more information or some other references that can help me?

**Comment by maciejkula** [comment]
As you've probably noticed, this is just to fool Keras.

**Comment by voladorlu** [comment]
Yes, I have noticed after a few study. It's a very smart way to make pairwise training available. However, situation can be very different if we wanna evaluate model with other top-N metrics like mean average precision, nDCG. Here, pairwise loss is naturally suitable for AUC, since it's based on globally pairwise comparisons. I think another model which shares weights can be defined as a point prediction value for a single (user, item) pair. It's a easy work for shallow model like matrix factorization, however, keras seems not provides a natural way to make it. Thanks very much for a response. ^ - ^
